### PR TITLE
Column IDs unique between scopes

### LINKED ISSUE #642: Column ID clashes when saving OutputDataStream jobs to XML
I am unable to reopen a job where there is both a (source table) column with id "col_city" and inside a child job there is another column with id "col_city".

The error I get is when loading/reading the job file, but it seems to me that the real cause is when it is being saved - it should not have ID clashes.

### PR DESCRIPTION
**Pull request by tomaszguzialek**
Fixes #642

### CODE REVIEW

File: engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobWriter.java [hunk 1]
```diff
@@ -162,7 +166,10 @@ private void configureJobType(final AnalysisJob analysisJob, final JobType jobTy
             sourceType.setDataContext(dataContextType);
         }
         // mappings for lookup of ID's
-        final BiMap<InputColumn<?>, String> columnMappings = HashBiMap.create(50);
+        if (columnMappings == null) {
+            columnMappings = HashBiMap.create(50);
```
Comment (kaspersorensen) [inline engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobWriter.java]: would be a bit nicer to have this in line 127 instead of the null.

**Comment by kaspersorensen** [comment]
LGTM... Only a minor cosmetical thingie

File: engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobWriter.java (hunk 1 above)
Comment (tomaszguzialek) [inline engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobWriter.java]: True, fixing now.

**Comment by kaspersorensen** [comment]
LGTM

**Pull request merged**